src/alloc/isolated_alloc.rs

```diff
@@ -255,18 +279,19 @@ impl IsolatedAlloc {
     /// SAFETY: Same as `dealloc()` with the added requirement that `layout`
     /// must ask for a size larger than the host pagesize.
     unsafe fn dealloc_huge(&mut self, ptr: *mut u8, layout: Layout) {
-        let layout = IsolatedAlloc::huge_normalized_layout(layout, self.page_size);
+        let size = self.huge_normalized_layout(layout, self.page_size);
         // Find the pointer matching in address with the one we got
         let idx = self
             .huge_ptrs
             .iter()
             .position(|pg| ptr.addr() == pg.0.addr())
             .expect("Freeing unallocated pages");
         // And kick it from the list
-        self.huge_ptrs.remove(idx);
-        // SAFETY: Caller ensures validity of the layout
+        let (un_offset_ptr, _) = self.huge_ptrs.remove(idx);
+        // SAFETY: huge_normalized_layout ensures the validity of the size, and
+        // we know the pointer was the original page-aligned one
```

Comment:
I have no idea what this safety comment is supposed to mean.
```suggestion
        // SAFETY: huge_ptrs contains allocations that we own, and huge_normalized_layout
        // computed the size of the original allocation (if the user gave us the right `layout`,
        // which we may assume).
```